### PR TITLE
TastyReader: enable opaque types and erase correctly

### PR DESCRIPTION
**Pull request by bishabosha**
This PR re-enables opaque types because they can remain opaque but be correctly erased.

The main part to observe here is the addition of another symbol attachment which stored the private alias of the opaque type which is recovered at erasure, perhaps there is a more efficient way to do this.

There may also be required some edge cases to check that the opaque type is always dealiased correctly.

### CODE REVIEW

File: src/reflect/scala/reflect/internal/transform/Erasure.scala [hunk 1]
```diff
@@ -151,7 +151,10 @@ trait Erasure {
         else if (sym.isDerivedValueClass) eraseDerivedValueClassRef(tref)
         else if (isDottyEnumSingleton(sym)) apply(intersectionType(tp.parents)) // TODO [tasty]: dotty enum singletons are not modules.
         else if (sym.isClass) eraseNormalClassRef(tref)
-        else apply(sym.info.asSeenFrom(pre, sym.owner)) // alias type or abstract type
+        else sym.attachments.get[OpaqueTypeAlias] match {
```
Comment (bishabosha) [inline src/reflect/scala/reflect/internal/transform/Erasure.scala]: Here we are using a symbol attachment to check if the symbol is an opaque type and extract its alias

**Review by dwijnand — APPROVED** [review]
Seems ok

**Comment by bishabosha** [comment]
Locally I've added a new test which is causing a compiler crash, so investigating

**Comment by bishabosha** [comment]
The case of opaque type alias with type parameters is now handled

**Review by dwijnand — APPROVED** [review]

**Pull request merged**